utils/pinwheels.py

```diff
@@ -0,0 +1,98 @@
+import pathlib
+import matplotlib.pyplot as plt
+import numpy as np
+import os 
+
+def plot_pinwheel(sample_id, final_scores, id, name):
+    """
+    Generates and saves a polar plot (pinwheel) visualization for pathway scores.
+
+    Parameters:
+        sample_id (str): Identifier for the sample being visualized.
+        final_scores (DataFrame): DataFrame containing pathway IDs and scores.
+        id (str): Column name for pathway identifiers.
+        name (str): Type of plot - e.g. pathway or drug.
+    """
+    save_dir = pathlib.Path("./visualize/pinwheels").resolve()
+    
+    # Get pathway names and scores
+    ids = final_scores[id].values
+    scores = final_scores["pathway_score"].values
+
+    # Convert to radians for plotting
+    num_pathways = len(ids)
+    angles = np.linspace(0, 2 * np.pi, num_pathways, endpoint=False).tolist()
+
+    # Create polar plot
+    fig, ax = plt.subplots(figsize=(8, 8), subplot_kw={"projection": "polar"})
+    ax.fill(angles, scores, color="mediumpurple", alpha=0.8)
+
+    # Get top N indices by score magnitude
+    top_n_indices = np.argsort(scores)[-5:]
```

Comment:
Would it be a good idea to make this a parameter? Instead of just having a hard coded number, maybe put a parameter into the function like `top_n_scores=5` and that will update this code too:
```suggestion
    top_n_indices = np.argsort(scores)[-top_n_scores:]
```

Reply:
yep!